File: no-auth/src/main/java/com/kjj/noauth/dto/user/UserDto.java
package com.kjj.noauth.dto.user;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.util.Random;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@ToString
public class UserDto {
    private Long id;
    private String username;
    private String password;
    private String roles;

    public static UserDto ofKeycloak(String sub, String roles) {
        StringBuilder sb = new StringBuilder();
        Random random = new Random();
        int length = 10;

        for (int i = 0; i < length; i++) {
            int digit = random.nextInt(10);
            sb.append(digit);
        }

        return new UserDto(
                null,
                sub,
                sb.toString(),
                roles
        );
    }
}

File: no-auth/src/main/kotlin/com/kjj/noauth/dto/user/UserDto.kt
package com.kjj.noauth.dto.user

import java.util.*

data class UserDto(
    val id: Long?,
    val username: String,
    val password: String,
    val roles: String
) {
    companion object {
        fun ofKeycloak(sub: String, roles: String): UserDto {
            val sb = StringBuilder()
            val random = Random()

            val length = 10
            for (i in 0 until length) sb.append(random.nextInt(10))

            return UserDto(
                null,
                sub,
                sb.toString(),
                roles
            )
        }
    }
}